### ISSUE TITLE
Create Dockerfile to run dev env

### ISSUE THREAD

**Issue by bazzadp** [HTTPArchive/almanac.httparchive.org#937]
As discussed in https://github.com/HTTPArchive/almanac.httparchive.org/issues/933#issuecomment-651280322 let's set up a Dockerfile to allow easier launching of dev env

**Comment by ibnesayeed** [comment]
What are latest recommended versions of Python and Node to be used to work with this repo?

**Comment by bazzadp** [comment]
Python v3 - should work with latest v3 version
Node v12 - should work with latest v12 version

**Comment by ibnesayeed** [comment]
When I ran `pytest` in the container, I saw the following two warnings:

```
====================================================== test session starts =======================================================
platform linux -- Python 3.8.3, pytest-5.4.3, py-1.9.0, pluggy-0.13.1
rootdir: /app
collected 12 items                                                                                                               

validate_test.py ............                                                                                              [100%]

======================================================== warnings summary ========================================================
validate.py:101
  /app/validate.py:101: DeprecationWarning: invalid escape sequence \s
    accepted_languages = re.findall('(?:^|\s|,)(\w+)', header)

/usr/local/lib/python3.8/site-packages/flask/sessions.py:14
  /usr/local/lib/python3.8/site-packages/flask/sessions.py:14: DeprecationWarning: Using or importing the ABCs from 'collections' instead of from 'collections.abc' is deprecated since Python 3.3, and in 3.9 it will stop working
    from collections import MutableMapping

-- Docs: https://docs.pytest.org/en/latest/warnings.html
================================================= 12 passed, 2 warnings in 1.38s =================================================
```

I can fix the first warning very easily by just making the RegEx a raw string, but the other warning will need update to Flask version. Any suggestions, what should we do about it?

**Comment by bazzadp** [comment]
No seen that first error before to be honest. Wonder if it’s due to the version you are running? But if you can see the fix then go for it.

I have seen the second and ignore it as not related to our code and just a warning. Maybe raise a separate issue for that one so we can look at it separately?

**Comment by ibnesayeed** [comment]
Implemented in #956 and is ready for reviews.
